contracts/ToucanRegenBridge.sol

```diff
@@ -79,27 +101,20 @@ contract ToucanRegenBridge is Ownable, Pausable {
      * This functions must be called by a bridge account.
      */
     function issueTCO2Tokens(
-        string memory sender,
+        string calldata sender,
         address recipient,
-        IToucanCarbonOffsets tco2,
+        address tco2,
         uint256 amount
-    ) public {
-        require(false, "Not implemented yet");
-        require(isRegenAddress(bytes(sender)), "recipient must a Regen Ledger account address");
+    ) external whenNotPaused isRegenAddress(bytes(sender)) {
+        require(amount > 0, "amount must be positive");
         require(msg.sender == regenBridge, "only bridge can issue tokens");
 
-        emit Issue(sender, recipient, address(tco2), amount);
-        // TODO: finish the implementation
-        // + mint tco2 tokens
-        // + define and implement checks
-    }
+        // Limit how many tokens can be minted per TCO2; this is going to underflow
+        // in case we try to mint more for a TCO2 than what has been burnt so it will
+        // result in reverting the transaction.
+        tco2Limits[tco2] -= amount;
 
-    function isRegenAddress(bytes memory recipient) internal pure returns (bool) {
-        // verification: checking if recipient starts with "regen1"
-        require(recipient.length >= 44, "regen address is at least 44 characters long");
-        bytes memory prefix = "regen1";
-        for (uint8 i = 0; i < 6; ++i)
-            require(prefix[i] == recipient[i], "regen address must start with 'regen1'");
-        return true;
+        emit Issue(sender, recipient, address(tco2), amount);
```

Comment:
is wrapping in address necessary here? `tco2` is already an address.

Reply:
Not needed, will remove